app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ActiveRecord::Base
+  # Method that updates the post counter for a given user.
+
+  def self.update_counter(user:)
```

Comment:
[Optional]
- [ ] You can user [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) to update the counter automatically after a new post created

```ruby
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
end
```